### ISSUE TITLE
TypeError when constructing a 204 Response from UploadFile#upload

### ISSUE THREAD

**Issue by nbenz** [adopted-ember-addons/ember-file-upload#809]
On versions >= 6.0.0 (#770 as far as I can tell) I'm receiving the following error as a response to a `file.upload(...)` call.
```
http-request.js:25 Uncaught TypeError: Failed to construct 'Response': Response with null body status cannot have body
```

I'm uploading said file to S3, and the reponse from AWS is a 204. However, when constructing the Response on [this line](https://github.com/adopted-ember-addons/ember-file-upload/blob/master/ember-file-upload/src/system/http-request.ts#L19), `request.response` evaluates to an empty string. Response expects either `null` or `undefined` as the first parameter when the response status is 204, hence the TypeError.

I'm not sure if the empty string is coming from AWS or simply the XMLHttpRequest#response method. It also seems like an issue that would have affected a lot of people already, so if I'm just missing something really obvious please let me know. I'm not doing anything special with the S3 upload, but I can provide a sample request if that is helpful.

This is exhibited on
* Chrome - Version 102.0.5005.115 (Build officiel) (arm64)
* Firefox - 101.0.1 (64-bit)
* Safari - Version 15.5 (17613.2.7.1.8)

using Ember Source 3.28

**Comment by gilest** [comment]
Thanks for reporting! This is hopefully fixed in 7.0.1

**Comment by nbenz** [comment]
I bumped to 7.0.2 today and everything is working. Thanks for such a quick turnaround!
